spec/support/shared_examples_for_recovery_codes.spec.rb

```diff
@@ -0,0 +1,44 @@
+shared_examples_for 'recovery code page' do
+  context 'regenerating recovery code with `Get another code` button' do
+    scenario 'displays a flash message and a new code' do
+      old_code = @user.reload.recovery_code
+
+      click_link t('users.recovery_code.get_another')
+
+      expect(@user.reload.recovery_code).to_not eq old_code
+      expect(page).to have_content t('notices.send_code.recovery_code')
+    end
+  end
+
+  context 'informational text' do
+    let(:accordion_selector) { generate_class_selector('accordion') }
+    let(:content_selector) { generate_class_selector('accordion-content') }
+
+    scenario 'it displays the recovery code info header' do
+      expect(page).to have_content(t('users.recovery_code.help_text_header'))
+    end
+
+    context 'with javascript disabled' do
+      scenario 'content is visible by default' do
+        expect(page).to have_xpath("//#{accordion_selector}[@aria-expanded='true']")
+        expect(page).to have_xpath("//#{content_selector}[@aria-hidden='false']")
+        expect(page).to have_content(t('users.recovery_code.help_text'))
+      end
+    end
+
+    context 'with javascript enabled', js: true do
```

Comment:
I can do some digging, but I think it may be difficult to make an accordion that is CSS only, but also a11y friendly to screen readers and keyboards.  I believe the USWDS accordion also relies on JS.  I tested @el-mapache's implementation out which shows the information in it's expanded state for non-JS users which seems like a good fallback option.